projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart-validation.e2e-spec.ts

```diff
@@ -0,0 +1,124 @@
+import { viewportContext } from '../../helpers/viewport-context';
+import { standardUser } from '../../sample-data/shared-users';
+import { addProductToCart } from '../../helpers/save-for-later';
+import {
+  lowStockResponse,
+  outOfStockResponse,
+  PRODUCT_1,
+  PRODUCT_2,
+} from '../../sample-data/cart-validation';
+import { removeCartItem } from '../../helpers/cart';
+
+context('Cart validation', () => {
+  viewportContext(['mobile', 'desktop'], () => {
+    beforeEach(() => {
+      cy.window().then((win) => win.sessionStorage.clear());
+      cy.window().then((win) => win.localStorage.clear());
+      cy.clearLocalStorageMemory();
+      cy.requireLoggedIn(standardUser);
+    });
+
+    describe('As logged in', () => {
+      beforeEach(() => {
+        cy.restoreLocalStorage();
+      });
+
+      afterEach(() => {
+        cy.saveLocalStorage();
+      });
+
+      after(() => {
+        removeCartItem(PRODUCT_1);
+      });
+
+      it('should display information about reduced stock for product in cart', () => {
+        addProductToCart(PRODUCT_1);
+        addProductToCart(PRODUCT_2);
+
+        cy.intercept(
+          {
+            method: 'POST',
+            pathname: `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
+              'BASE_SITE'
+            )}/users/current/carts/*/validate`,
+          },
+          lowStockResponse
+        ).as('validate');
+
+        cy.findByText(/proceed to checkout/i).click();
+        cy.wait(`@validate`);
+
+        cy.get('cx-global-message').should(
```

Comment:
lines 51-54 exist in this file a few times, I will suggest to put it in helper method to reduce code